server/src/test/java/org/elasticsearch/index/query/QueryStringQueryBuilderTests.java

```diff
@@ -1290,6 +1299,61 @@ public void testQuoteAnalyzer() throws Exception {
         assertEquals(expectedQuery, query);
     }
 
+    // The only expectation for this test is to not throw exception
+    public void testToQueryExceedingMaxClauseCountWithUnmappedFields() throws Exception {
+        QueryShardContext shardContext = createShardContext();
+
+        new QueryStringQueryBuilder("\"bar\"")
+                .field("*")
+                .field("unmappedField1")
+                .field("unmappedField2")
+                .field("unmappedField3")
+                .field("unmappedField4")
+                .toQuery(shardContext);
+    }
+
+    public void testToQueryExceedingMaxCaluseCountWithMappedFeilds() throws IOException {
+        QueryShardContext shardContext = createShardContext();
+        shardContext.getMapperService()
+            .merge("_doc", new CompressedXContent(Strings.toString(PutMappingRequest.buildFromSimplifiedDef("_doc",
+                "additionalMappedField1", "type=text",
+                "additionalMappedField2", "type=text",
+                "additionalMappedField3", "type=text",
+                "additionalMappedField4", "type=text"
+            ))), MapperService.MergeReason.MAPPING_UPDATE);
+
+        try {
+            new QueryStringQueryBuilder("\"bar\"")
+                .field("*")
+                .toQuery(shardContext);
+            fail();
+        }
+        catch (IllegalArgumentException e) {
+            assertEquals("field expansion matches too many fields, limit: 13, got: 17", e.getMessage());
+        }
+
+    }
+
+    public void testToQueryExceedingMaxCaluseCountWithMappedFeildsWithoutWeightAndBoost() throws IOException {
```

Comment:
lot of words ;), 
```suggestion
    public void testToQueryMaxClauseNoWeight() throws IOException {
```
?

Reply:
Done.